user-management/src/main/java/de/tum/in/www1/artemis/usermanagement/service/user/UserService.java

```diff
@@ -0,0 +1,594 @@
+                }
+                else {
+                    log.info("Create internal admin user {}", artemisInternalAdminUsername.get());
+                    ManagedUserVM userDto = new ManagedUserVM();
+                    userDto.setLogin(artemisInternalAdminUsername.get());
+                    userDto.setPassword(artemisInternalAdminPassword.get());
+                    userDto.setActivated(true);
+                    userDto.setFirstName("Administrator");
+                    userDto.setLastName("Administrator");
+                    userDto.setEmail("admin@localhost");
+                    userDto.setLangKey("en");
+                    userDto.setCreatedBy("system");
+                    userDto.setLastModifiedBy("system");
+                    // needs to be mutable --> new HashSet<>(Set.of(...))
+                    userDto.setAuthorities(new HashSet<>(Set.of(ADMIN.getAuthority(), STUDENT.getAuthority())));
+                    userDto.setGroups(new HashSet<>());
+                    userCreationService.createUser(userDto);
+                }
+            }
+        }
+        catch (Exception ex) {
+            log.error("An error occurred after application startup when creating or updating the admin user or in the LDAP search", ex);
+        }
+    }
+
+    /**
+     * Activate user registration
+     *
+     * @param key activation key for user registration
+     * @return user if user exists otherwise null
+     */
+    public Optional<User> activateRegistration(String key) {
+        log.debug("Activating user for activation key {}", key);
+        return userRepository.findOneWithGroupsByActivationKey(key).map(user -> {
+            activateUser(user);
+            return user;
+        });
+    }
+
+    /**
+     * Activates the user and cancels the automatic cleanup of the account.
+     *
+     * @param user the non-activated user
+     */
+    public void activateUser(User user) {
+        // Cancel automatic removal of the user since it's activated.
+        instanceMessageSendService.sendCancelRemoveNonActivatedUserSchedule(user.getId());
+        optionalVcsUserManagementService.ifPresent(vcsUserManagementService -> vcsUserManagementService.activateUser(user.getLogin()));
+        // activate given user for the registration key.
+        userCreationService.activateUser(user);
+    }
+
+    /**
+     * Reset user password for given reset key
+     *
+     * @param newPassword new password string
+     * @param key         reset key
+     * @return user for whom the password was performed
+     */
+    public Optional<User> completePasswordReset(String newPassword, String key) {
+        log.debug("Reset user password for reset key {}", key);
+        return userRepository.findOneByResetKey(key).filter(user -> user.getResetDate().isAfter(Instant.now().minusSeconds(86400))).map(user -> {
```

Comment:
(just a suggestion) this specific number (86400) can be defined as constant instead directly passing as a parameter.